### PR TITLE
feature(#77): Mandate List frontend

### LINKED ISSUE #77: [core] Implement UI as list over mandates
## Short description of the feature
A page where all current mandates can be looked up.

Depends on #70

[x] Relates to frontend
[ ] Relates to the backend
[ ] Requires modification of GraphQL
[ ] Requires modification of Database
[ ] Requires creating a new service

### PR DESCRIPTION
**Pull request by kuleshmaria**
closes #77

### CODE REVIEW

File: frontend/components/Mandates/MandateSet.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  return (
+    <TableCell align="left">
+        {
+          mandates.map( (mandate) => (mandate) ? (
+            <Link href={routes.member(mandate.member.id)}>
+              <p>{mandate.member.first_name} {mandate.member.last_name}</p>
+            </Link>
+          )
+            : (<div>No mandates were found for this position.</div>)
```
Comment (oliverlevay) [inline frontend/components/Mandates/MandateSet.tsx:18]: Maybe has to be translated?

File: frontend/pages/mandates/[year].tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      <>
+        <h2 className="classes.positionName">{ t('mandates') } {year}</h2>
+          {
+            (lthOpens <= parseInt(year) && parseInt(year) <= currentYear) ? (
+              <>
+              <Stepper moveForward={moveForward} moveBackward={moveBackward} year={parseInt(year)} idx={timeInterval-(parseInt(year)-lthOpens)} maxSteps={timeInterval}></Stepper>
+              <MandateList year={year} />
+              </>
+            )
+              : (<div>No mandates were found for this year.</div>)
```
Comment (oliverlevay) [inline frontend/pages/mandates/[year].tsx:38]: Needs to be translated

File: backend/services/core/src/schema.graphql [hunk 3]
```diff
@@ -62,6 +63,15 @@ type Mandate @key(fields: "id") {
   member: Member
 }
 
+type MandatesByPosition {
+  mandate: Mandate,
```
Comment (emilwihlander) [inline backend/services/core/src/schema.graphql:67]: Am I crazy or should this be a Position?

Comment (kuleshmaria) [reply backend/services/core/src/schema.graphql]: fixed

File: frontend/components/Mandates/MandateList.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    });
+    return map;
+  }
+
+  const mandateList = data.mandates.mandates;
+  const mandatesByPosition = groupBy(mandateList, e => t(e.position.name), e => e.member);
+  const positions = Array.from(mandatesByPosition
+                                .keys())
+                                .sort((a, b) => a.localeCompare(b));
+
```
Comment (emilwihlander) [inline frontend/components/Mandates/MandateList.tsx:55]:
```typescript
if (!hasAccess(apiContext, 'core:mandate:read'))
  return <></>
```

File: frontend/components/Mandates/Stepper.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { useTheme } from '@emotion/react';
+import { KeyboardArrowRight, KeyboardArrowLeft } from '@mui/icons-material';
+import { Box, MobileStepper, Button } from '@mui/material';
+import * as React from 'react';
+import { mandateStyles } from './mandatestyles';
+
+export default function Stepper({ moveForward, moveBackward, year, idx, maxSteps}) {
+  const theme = useTheme();
+
+  const [activeStep, setActiveStep] = React.useState(idx);
```
Comment (emilwihlander) [inline frontend/components/Mandates/Stepper.tsx:10]:
```typescript
import React, { useState } from 'react';
// ...
const [activeStep, setActiveStep] = useState(idx);
```

File: frontend/pages/mandates/[year].tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    router.push("/mandates/"+(parseInt(year)-1))
+  };
+
+  const moveBackward = () => {
+    router.push("/mandates/"+(parseInt(year)+1))
+  };
+
+  return (
+    <DefaultLayout>
+      <>
```
Comment (emilwihlander) [inline frontend/pages/mandates/[year].tsx:29]: This seems redundant?

File: frontend/public/locales/en/mandate.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  "Dagsansvarig": "Cafe manager",
+  "DWWW-ansvarig": "Head of DWWW",
+  "Fotograf": "Photographer",
+  "Funktionär inom Skattmästeriet": "Treasurer Volunteer",
+  "sudo": "Sudo",
+  "Tandemgeneral": "Tandem Relay Team Leader",
+  "Nollningsfunktionär": "Introductions Volunteer",
+  "Vinförman": "Wine Warden",
+  "Sektionskock": "Guild Cook",
+  "Skattmästare": "Treasurer",
```
Comment (emilwihlander) [inline frontend/public/locales/en/mandate.json:15]: This data should be fetched from the backend

File: frontend/public/locales/sv/mandate.json [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  "Dagsansvarig": "Dagsansvarig",
+  "DWWW-ansvarig": "DWWW-ansvarig",
+  "Fotograf": "Fotograf",
+  "Funktionär inom Skattmästeriet": "Funktionär inom Skattmästeriet",
+  "sudo": "Sudo",
+  "Tandemgeneral": "Tandemgeneral",
+  "Nollningsfunktionär": "Nollningsfunktionär",
+  "Vinförman": "Vinförman",
+  "Sektionskock": "Sektionskock",
+  "Skattmästare": "Skattmästare",
```
Comment (emilwihlander) [inline frontend/public/locales/sv/mandate.json:15]: Same here

File: frontend/components/Mandates/MandateList.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+         const key = keyGetter(item);
+         const x = valueGetter(item);
+         if (!map.has(key)) {
+            map.set(key, [x]);
+         } else {
+            map.get(key).push(x);
+         }
+    });
+    return map;
+  }
```
Comment (LucasBoberg) [inline frontend/components/Mandates/MandateList.tsx:50]: Should be moved to the functions folder in frontend

File: frontend/components/Navigation/NavigationItems.tsx [hunk 10]
```diff
@@ -6,14 +6,29 @@ import LibraryBooksIcon from '@mui/icons-material/LibraryBooks';
...
 
 const navigationItems: NavigationItem[] = [
   {
     translationKey: 'home',
     path: routes.root,
     icon: <HomeIcon color="primary" />,
   },
+  {
+    translationKey: "D-sek",
+    path: routes.dsek,
```
Comment (LucasBoberg) [inline frontend/components/Navigation/NavigationItems.tsx:22]: Change to "Sektionen/The Guild" istead of D-sek

File: frontend/public/locales/sv/mandate.json [hunk 11]
```diff
@@ -0,0 +1,7 @@
+{
+  "mandates": "Funktionärer",
+  "positions": "Poster",
+  "mandateError": "Inga mandat hittade för det här året.",
+  "current mandates": "Aktiva Funktionärer",
+  "vakant": "V.A.Cant"
```
Comment (LucasBoberg) [inline frontend/public/locales/sv/mandate.json:6]: Think this is revesed

File: frontend/public/locales/en/mandate.json [hunk 12]
```diff
@@ -0,0 +1,7 @@
+{
+  "mandates": "Mandates",
+  "positions": "Positions",
+  "mandateError": "No mandates were found for this year.",
+  "current mandates": "Current Mandates",
+  "vakant": "V.A.Kant"
```
Comment (LucasBoberg) [inline frontend/public/locales/en/mandate.json:6]: Think this is revesed

**Review by LucasBoberg — CHANGES_REQUESTED** [review]
See comments

File: frontend/components/Navigation/NavigationItems.tsx [hunk 13]
```diff
@@ -18,12 +18,12 @@ const navigationItems: NavigationItem[] = [
     icon: <HomeIcon color="primary" />,
   },
   {
-    translationKey: "D-sek",
+    translationKey: "dsek",
```
Comment (LucasBoberg) [inline frontend/components/Navigation/NavigationItems.tsx:21]: should be a diffrent name on the key

File: frontend/public/locales/en/common.json [hunk 14]
```diff
@@ -1,5 +1,6 @@
 {
   "home": "Home",
+  "dsek": "The Guild",
```
Comment (LucasBoberg) [inline frontend/public/locales/en/common.json:3]: should be a diffrent name on the key

File: frontend/public/locales/sv/common.json [hunk 15]
```diff
@@ -1,13 +1,14 @@
 {
   "home": "Hem",
+  "dsek": "Sektionen",
```
Comment (LucasBoberg) [inline frontend/public/locales/sv/common.json:3]: should be a diffrent name on the key

**Review by LucasBoberg — CHANGES_REQUESTED** [review]
See comments

**Review by LucasBoberg — APPROVED** [review]

**Pull request merged**